### ISSUE TITLE
Media Library: Can't migrate to Azure Blob Storage automatically

### ISSUE THREAD

**Issue by BenedekFarkas** [OrchardCMS/Orchard#6756]
Let's consider the following scenario:

- I'm developing an Orchard application locally while using the regular file storage (Orchard.Web\Media) for media files.
- After deploying the application to Azure, I also want to use Blob Storage:
    - Enable the Azure Media Storage Feature.
    - Copy all files from the local file system to the Storage Account.

I would expect the Media Library to continue to work as before, but media files are not loading at all for any folder, except "Recent". That's because Orchard stores the path of the media files based on their location on the file system (at the time they are created): NTFS uses `\` as the directory separator, but AFS uses `/`, so when querying for content items in a specific folder while using AFS won't return any results for media items created on NTFS (and vice versa).

My suggestion is that Media Library should always use one of those (I don't think it matters which one, since both of them are illegal characters for file and folder names).

**Comment by hannan-azam** [comment]
@BenedekFarkas i agree with you, there should be a simple solution to migrate to Azure Blob/File System anytime one decides to go with either storage.

I came across this problem working on a project where the site involved around 6000 media items. We kept working happily with local file system and once we were ready to go live with Azure Blob Storage we got struck with this. Later we solved it by running SQL Queries to convert all those directory separators to "/".

In HTML Content Editors and wherever media urls are used, Orchard uses URL based paths so  "/" separator would be a good candidate for this.

**Comment by sebastienros** [comment]
I agree, the database should have normalized file paths, using '/' might be the simplest.

**Comment by rtpHarry** [comment]
I'm using 1.10.x with Azure Media Storage module enabled and that new bit of code is failing:

![image](https://cloud.githubusercontent.com/assets/1038062/14753954/709f406c-08d0-11e6-93fc-ad942e461e08.png)

![image](https://cloud.githubusercontent.com/assets/1038062/14754061/e7478f12-08d0-11e6-9289-e55a190982a3.png)

It uploaded my kitty test pictures (I can see them in the blob) but now I can't view them.

## Repro
This is an almost out of the box setup, I have cloned 1.10.x at 'b8e138b2d55734c9f6257d08fa944fb48b1676cb' then set up the azure settings and media to be stored in blob. Created a folder called 'cat pictures', dropped three pictures into it and tried to go back to media browser but all pages (folder page and recent page) are showing the display error.

**Comment by rtpHarry** [comment]
Hmm actually it only happens after I have uploaded and clicked "close".

When I navigate away from it (for example clicking on to Widgets admin page) and then click back to Media, everything works as expected.
